page-objects/administrateClasses.js

```diff
@@ -0,0 +1,366 @@
+        let upperContainer = await driver.$('.container-fluid.ajaxcontent > div:nth-child(3)');
+        let container = await upperContainer.$('.chosen-container');
+        const searchInput = await container.$(".chosen-search-input");
+        await searchInput.click();
+        await searchInput.setValue(searchName);
+        const searchResult = await container.$(`.chosen-results .active-result.red`)
+        await searchResult.click();
+        await this.submitAddStudentTotheClass();
+    },
+    submitAddStudentTotheClass: async function() {
+        let btnContainer = await driver.$('.container-fluid.ajaxcontent');
+        let btn = await btnContainer.$('button[type="submit"]');
+        await btn.click();
+        let selectorToBeLoaded = await driver.$('.container-fluid.ajaxcontent tr');
+        await selectorToBeLoaded.waitForExist(3000);
+    },
+    // get all classes names given we are on classes administration page
+    getAllClassNames: async function() {
+        await this.deleteFilterSchoolYears();
+        let isThereAnyClass = await this.isThereAnyClass();
+        let isPaginated = await this.isPaginated();
+        let names = [];
+        if (isPaginated==true ) {
+            let pagesSelector = await driver.$$('.pagination-wrapper .pagination li');
+            let lastpageIndex = pagesSelector.length-2;
+            for (var i=3; i<=lastpageIndex; i++) {
+                let namesContainer = await driver.$('[data-testid="students_names_container"]');
+                let allClassesOnThePage = await namesContainer.$$('tr');
+                for(var j=1; j<=allClassesOnThePage.length; j++) {
+                    let row = await namesContainer.$('tr:nth-child('+j+')');
+                    let nameSelector = await row.$('td:nth-child(1)');
+                    let name = await nameSelector.getText();
+                    await names.push(name);
+                }
+                if (i==lastpageIndex) {
+                    return names; 
+                } else { 
+                    let nextPageSelector = await driver.$('.pagination-wrapper .pagination li:nth-child('+(i+1)+') > a');
+                    await nextPageSelector.click()
+                    await driver.pause(1500);
+                }
+            };
+        } else if (isThereAnyClass==true && isPaginated==false) {
+            let names = [];
+            let namesContainer = await driver.$('[data-testid="students_names_container"]');
+            let allClasses = await namesContainer.$$('tr');
+            for(var i=1; i<=allClasses.length; i++) {
+                let row = await namesContainer.$('tr:nth-child('+i+')');
+                let nameSelector = await row.$('td:nth-child(1)');
+                let name = await nameSelector.getText();
+                await names.push(name);
+        }
+    
+        return names;
+        
+        } else {
+            return "there are no classes";
```

Comment:
It seems you are mixing return types here:
- if there are classes, you return an array
- if there are no classes, you return a string

This can be very counter-intuitive to callers of the method. Consider returning an empty array in the second case or throwing an exception if this is unexpected